### PR TITLE
IGNITE-18674 Introduce async logic to CLI

### PR DESCRIPTION
**Pull request by PakhomovAlexander**
I've introduced the core functionality for background async calls with a progress bar. The SQL completion hanging is fixed with background async initialization. 

I don't see any reason to use a UI thread with a special API.

https://issues.apache.org/jira/browse/IGNITE-18674

### CODE REVIEW

File: modules/cli/src/main/java/org/apache/ignite/internal/cli/core/call/AsyncCallExecutionPipeline.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            return handleResult(result);
+        } catch (CompletionException e) {
+            return handleException(e.getCause());
+        } catch (Exception e) {
+            return handleException(e);
+        }
+    }
+
+    private void print(String s) {
+        output.print("\r" + s);
```
Comment (valepakh) [inline modules/cli/src/main/java/org/apache/ignite/internal/cli/core/call/AsyncCallExecutionPipeline.java:79]: Could you please add a comment explaining this?

File: modules/cli/src/main/java/org/apache/ignite/internal/cli/commands/sql/SqlCompleter.java [hunk 2]
```diff
@@ -72,4 +73,9 @@ private static void addCandidate(String string, List<Candidate> candidates) {
         candidates.add(new Candidate(string));
         candidates.add(new Candidate(string.toLowerCase()));
     }
+
+    public void initStateAsync() {
+        // trigger schema loading in background
+        CompletableFuture.supplyAsync(schemaProvider::getSchema);
```
[the remark this answers is not in the document]
Comment (PakhomovAlexander) [reply modules/cli/src/main/java/org/apache/ignite/internal/cli/commands/sql/SqlCompleter.java:79]: Probably yes

File: modules/cli/src/main/java/org/apache/ignite/internal/cli/core/call/AsyncCallExecutionPipeline.java (hunk 1 above)
Comment (PakhomovAlexander) [reply modules/cli/src/main/java/org/apache/ignite/internal/cli/core/call/AsyncCallExecutionPipeline.java:79]: Sure

**Comment by valepakh** [comment]
LGTM

**Review by valepakh — APPROVED** [review]

**Pull request merged**